src/thing_physics.c

```diff
@@ -61,14 +62,196 @@ TbBool thing_touching_flight_altitude(const struct Thing *thing)
         && (thing->mappos.z.val <= floor_height + 19*NORMAL_FLYING_ALTITUDE/17);
 }
 
-void slide_thing_against_wall_at(struct Thing *thing, struct Coord3d *pos, long a3)
+void slide_thing_against_wall_at(struct Thing *thing, struct Coord3d *pos, long blocked_flags)
 {
-    _DK_slide_thing_against_wall_at(thing, pos, a3); return;
+   // _DK_slide_thing_against_wall_at(thing, pos, a3); return;
+  unsigned short x_thing;
+  unsigned short sizexy;
+  unsigned short x_pos;
+  unsigned short y_thing;
+  unsigned short y_pos;
+  switch ( blocked_flags )
+  {
+    case SlbBloF_WalledX:
+      x_thing = *(unsigned short *)&thing->mappos.x.stl;
+      sizexy = (unsigned short)actual_sizexy_to_nav_sizexy_table[thing->clipbox_size_xy] >> 1;
+      x_pos = *(unsigned short *)&pos->x.stl;
```

Comment:
All these casts are avoidable if you pick the right union member.

Reply:
From what I can tell, the original code uses both unsigned char members as if it were 1 unsigned short.

Reply:
There is i.e. `pos->x.stl.val`

Reply:
The decompiled code takes a word at the address of stl.pos, so it looks to me like it's taking both stl.pos and stl.num in one go.

Reply:
pos->x.stl is a union so it is two thing at same address. And stl.val is whole thing as one unsigned short number.

Reply:
So you reckon the pointer *(unsigned __int16 *)&thing->mappos.x_stl_pos actually refers to mappos.x.val?

Reply:
What about bounce_thing_off_wall_at?